src/test/ui/lint/lint-deref-nullptr.rs

```diff
@@ -2,29 +2,37 @@
 
 #![deny(deref_nullptr)]
 
+use std::ptr;
+
+struct Struct {
+    field: u8,
+}
+
 fn f() {
     unsafe {
         let a = 1;
         let ub = *(a as *const i32);
         let ub = *(0 as *const i32);
         //~^ ERROR dereferencing a null pointer
-        let ub = *core::ptr::null::<i32>();
+        let ub = *ptr::null::<i32>();
+        //~^ ERROR dereferencing a null pointer
+        let ub = *ptr::null_mut::<i32>();
         //~^ ERROR dereferencing a null pointer
-        let ub = *core::ptr::null_mut::<i32>();
+        let ub = *(ptr::null::<i16>() as *const i32);
         //~^ ERROR dereferencing a null pointer
-        let ub = *(core::ptr::null::<i16>() as *const i32);
+        let ub = *(ptr::null::<i16>() as *mut i32 as *mut usize as *const u8);
         //~^ ERROR dereferencing a null pointer
-        let ub = *(core::ptr::null::<i16>() as *mut i32 as *mut usize as *const u8);
+        let ub = &*ptr::null::<i32>();
         //~^ ERROR dereferencing a null pointer
-        let ub = &*core::ptr::null::<i32>();
+        ptr::addr_of!(*ptr::null::<i32>());
         //~^ ERROR dereferencing a null pointer
-        core::ptr::addr_of!(*core::ptr::null::<i32>());
+        ptr::addr_of_mut!(*ptr::null_mut::<i32>());
         //~^ ERROR dereferencing a null pointer
-        std::ptr::addr_of_mut!(*core::ptr::null_mut::<i32>());
+        let ub = *ptr::null::<i32>();
```

Comment:
Previously you were testing both the `core` and `std` versions of these functions (which makes sense, making sure they are both recognized correctly as creating NULL pointers) -- but now this here seems just redundant with the test on line 17, it does the exact same thing.

Reply:
right should I remove the test or readd the core / std path ?

Reply:
Either way is fine for me.